src/views/MediaView/index.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect, useReducer, useCallback } from 'react'
+import FilterBar from '../../components/FilterBar'
+import MediaCard from '../../components/MediaCard'
+import Grid from '@mui/material/Unstable_Grid2'
+import Box from '@mui/material/Box'
+import NoResultsMessage from '../../components/NoResultsMessage'
+import fetchMedia from './utils/fetchMedia'
+import { FetchMediaResponse, MediaItem } from '../../types/interfaces/MediaData'
+import {
+  MediaFilterDispatchParams,
+  MediaFilterState,
+} from '../../types/interfaces/MediaFiltersReducer'
+import Backdrop from '@mui/material/Backdrop'
+import CircularProgress from '@mui/material/CircularProgress'
+
+const initialMediaFilterState: MediaFilterState = {
+  years: [],
+  genres: [],
+  searchText: '',
+  type: '',
+}
+
+const mediaFiltersReducer = (
+  state: MediaFilterState,
+  action: MediaFilterDispatchParams,
+): MediaFilterState => {
+  switch (action.type) {
+    case 'SET_YEARS':
+    case 'SET_GENRES':
+    case 'SET_SEARCH_TEXT':
+    case 'SET_TYPE':
+      return { ...state, ...action.payload }
+    case 'CLEAR_FILTERS':
+      console.log('clearing')
+      return {
+        ...initialMediaFilterState,
+      }
+    default:
+      return state
+  }
+}
+
+const MediaView = () => {
+  const [state, dispatch] = useReducer(
+    mediaFiltersReducer,
+    initialMediaFilterState,
+  )
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState<string | null>(null)
+  const [mediaData, setMediaData] = useState<MediaItem[] | null>(null)
+
+  useEffect(() => {
+    let ignore = false
+    console.log('state changed', state)
+    const handleFetchMediaData = async () => {
+      setLoading(true)
+      setError(null)
+
+      const { error, data }: FetchMediaResponse = await fetchMedia(state)
+
+      if (ignore) {
+        return
+      } else if (error) {
+        setError(error)
+      } else {
+        setMediaData(data)
+      }
+
+      setLoading(false)
+    }
+
+    handleFetchMediaData()
+
+    return () => {
+      ignore = true
+    }
+  }, [state])
+
+  const memoizedDispatch: ({
```

Comment:
The dispatch function returned by `useReducer` _should_ stay referentially equal between re-renders, so you probably don't need this.

Reply:
Yes! i think so too. This was a hail mary when I was troublehsooting the loading issue. 

Interesting enough, the react docs don't sepciy whether the dispatch function is natively memoized

Reply:
Oops, I didn't actually check until you said that. Good point! I still think that's the case similar to the `setState` callback given from `useState`, but could definitely be wrong.